packages/terra-hookshot/src/Hookshot.jsx

```diff
@@ -0,0 +1,336 @@
+        if (parent !== target.ownerDocument) {
+          parent.removeEventListener('scroll', this.tick);
+        }
+      });
+    }
+    this.listenersAdded = false;
+  }
+
+  position(event, resetCache) {
+    if (resetCache) {
+      this.cachedRects = this.getNodeRects();
+    } else {
+      this.cachedRects.targetRect = HookshotUtils.getBounds(this.props.targetRef());
+    }
+
+    const result = HookshotUtils.positionStyleFromBounds(
+      this.cachedRects.boundingRect,
+      this.cachedRects.targetRect,
+      this.cachedRects.contentRect,
+      this.cOffset,
+      this.tOffset,
+      this.cAttachment,
+      this.tAttachment,
+      this.props.attachmentMargin,
+      this.props.attachmentBehavior,
+    );
+
+    let styleUpdated = false;
+    if (this.contentNode.style.position !== result.style.position) {
+      this.contentNode.style.position = result.style.position;
+      styleUpdated = true;
+    }
+    const newTransform = `translate3d(${result.style.left}, ${result.style.top}, 0px)`;
+    if (this.contentNode.style.transform !== newTransform) {
+      this.contentNode.style.transform = newTransform;
+      styleUpdated = true;
+    }
+    if (this.contentNode.style.opacity !== '1') {
+      this.contentNode.style.opacity = '1';
+      styleUpdated = true;
+    }
+    if (styleUpdated) {
+      this.cachedRects.contentRect = HookshotUtils.getBounds(this.contentNode);
+    }
+
+    if (this.props.onPosition) {
+      this.props.onPosition(
+        event,
+        this.cachedRects.targetRect,
+        this.cachedRects.contentRect,
+        result.positions.cCoords.attachment,
+        result.positions.tCoords.attachment,
+        result.positions.tCoords.offset,
+      );
+    }
+  }
+
+  destroy() {
+    this.disableListeners();
+    this.contentNode = null;
+  }
+
+  update(event) {
+    if (!this.props.targetRef() || !this.contentNode) {
+      return;
+    }
+    this.updateHookshot(event);
+  }
+
+  updateHookshot(event) {
+    const resetCache = !event || (event.type !== 'scroll' && event.type !== 'touchmove');
+    this.position(event, resetCache);
+  }
+
+  cloneContent(content) {
+    return React.cloneElement(content, { refCallback: this.wrappedRefCallback(content) });
+  }
+
+  wrappedRefCallback(content) {
+    const initialRefCallback = content.props.refCallback;
+
+    return (node) => {
+      this.setContentNode(node);
+
+      if (initialRefCallback) {
+        initialRefCallback(node);
+      }
+    };
+  }
+
+  render() {
+    /* eslint-disable no-unused-vars */
+    const {
+      attachmentBehavior,
+      attachmentMargin,
+      boundingRef,
+      content,
+      contentAttachment,
+      contentOffset,
+      isEnabled,
+      isOpen,
+      targetRef,
+      targetAttachment,
+      targetOffset,
+      onPosition,
+    } = this.props;
+    /* eslint-enable no-unused-vars */
+    if (!isOpen) {
+      return null;
+    }
+
+    this.cOffset = HookshotUtils.parseOffset(contentOffset);
```

Comment:
Can we not abbreviate these?  It's tough when looking elsewhere in the code that this is content and target offset/attachment/etc.

Reply:
I mainly abbreviated them to display a difference between the parsed and unparsed values. If you have a recommendation I'd be fine changing them.

Reply:
I'm either ok with this or I am wondering if we like something like:

```
this.content = {
  offset: <blah>
  attachment: <blah>
}
```